bench/workgen/workgen.cxx

```diff
@@ -480,42 +469,89 @@ int Monitor::run() {
+        double interval_secs = options->sample_interval_ms / 1000.0;
+        
+        // Format entry into _out stream.
+        if (_out != NULL)
+            _format_out_entry(&interval, interval_secs, &t, checkpointing, tm);
+
+        // Format entry into _json stream.
+        if (_json != NULL)
+            _format_json_entry(tm, &t, first_iteration, &interval, checkpointing, interval_secs);
+
+        // Check latency threshold. Write warning into std::cerr in case read, inser or update
+        // exceeds latency_max.
+        _check_latency_threshold(&interval, latency_max);
+
+        prev_interval.assign(interval);
+        prev_totals.assign(new_totals);
+
+        first_iteration = false;
+    }
+
+    // Format JSON suffix.
+    if (_json != NULL)
+        _format_json_suffix();
+
+    return (0);
+}
 
-        uint64_t totalsec = ts_sec(t - _wrunner._start);
-        (*_out) << time_buf
-                << "," << totalsec
-                << "," << cur_reads
-                << "," << cur_inserts
-                << "," << cur_updates
-                << "," << (checkpointing ? 'Y' : 'N')
-                << "," << interval.read.average_latency()
-                << "," << interval.read.min_latency
-                << "," << interval.read.max_latency
-                << "," << interval.insert.average_latency()
-                << "," << interval.insert.min_latency
-                << "," << interval.insert.max_latency
-                << "," << interval.update.average_latency()
-                << "," << interval.update.min_latency
-                << "," << interval.update.max_latency
-                << std::endl;
-
-        if (_json != NULL) {
-#define    WORKGEN_TIMESTAMP_JSON        "%Y-%m-%dT%H:%M:%S"
-            buf_size = strftime(time_buf, sizeof(time_buf),
-              WORKGEN_TIMESTAMP_JSON, tm);
-            ASSERT(buf_size <= sizeof(time_buf));
-            snprintf(&time_buf[buf_size], sizeof(time_buf) - buf_size,
-              ".%3.3" PRIu64 "Z", (uint64_t)ns_to_ms(t.tv_nsec));
-
-            // Note: we could allow this to be configurable.
-            int percentiles[4] = {50, 95, 99, 0};
+void Monitor::_format_out_header() {
+    (*_out) << "#time,"
+            << "totalsec,"
+            << "read ops per second,"
+            << "insert ops per second,"
+            << "update ops per second,"
+            << "checkpoints,"
+            << "read average latency(uS),"
+            << "read minimum latency(uS),"
+            << "read maximum latency(uS),"
+            << "insert average latency(uS),"
+            << "insert min latency(uS),"
+            << "insert maximum latency(uS),"
+            << "update average latency(uS),"
+            << "update min latency(uS),"
+            << "update maximum latency(uS)"
+            << std::endl;
+}
+
+void Monitor::_format_out_entry(Stats *pinterval, double interval_secs, timespec *ptimespec, bool checkpointing, tm *tm) {
```

Comment:
I would avoid pointers where unnecessary and use references instead, i.e const Stats &interval instead.

Reply:
Same for other methods

Reply:
Done, thank you!